nilearn/maskers/nifti_labels_masker.py

```diff
@@ -307,15 +317,7 @@ def _post_masking_atlas(self, visualize=False):
 
         return masked_atlas, removed_region_ids, removed_region_names, display
 
-    def generate_report(self):
-        """Generate a report."""
-        from nilearn.reporting.html_report import generate_report
-
-        self._init_report_content()
-
-        return generate_report(self)
-
-    def _reporting(self):
+    def _get_displays(self):
```

Comment:
Renamed `_reporting` in maskers as `_get_displays` and moved `_reporting` inside `ReportingMixin` to do common checks and then call `_get_displays`.